### PR TITLE
Support custom IAM roles for cluster and workers

### PR DESCRIPTION
**Pull request by erks**
# PR o'clock

## Description

Fixes https://github.com/terraform-aws-modules/terraform-aws-eks/issues/282

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/eks_test_fixture` directories (look in CI for an example)
- [ ] Tests for the changes have been added and passing (for bug fixes/features)
- [x] Test results are pasted in this PR (in lieu of CI)
- [x] I've added my change to CHANGELOG.md
- [ ] Any breaking changes are highlighted above

### CODE REVIEW

**Comment by max-rocket-internet** [comment]
Hey @erks ! Thanks for the PR and sorry for the delay. I will review this PR soon. It's quite complex so taking me a little longer to find the time 🙂

**Comment by erks** [comment]
@max-rocket-internet thanks for taking a look!

File: workers.tf [hunk 1]
```diff
@@ -38,7 +38,7 @@ resource "aws_launch_configuration" "workers" {
   name_prefix                 = "${aws_eks_cluster.this.name}-${lookup(var.worker_groups[count.index], "name", count.index)}"
   associate_public_ip_address = "${lookup(var.worker_groups[count.index], "public_ip", local.workers_group_defaults["public_ip"])}"
   security_groups             = ["${local.worker_security_group_id}", "${var.worker_additional_security_group_ids}", "${compact(split(",",lookup(var.worker_groups[count.index],"additional_security_group_ids", local.workers_group_defaults["additional_security_group_ids"])))}"]
-  iam_instance_profile        = "${element(aws_iam_instance_profile.workers.*.id, count.index)}"
+  iam_instance_profile        = "${element(coalescelist(aws_iam_instance_profile.workers.*.id, data.aws_iam_instance_profile.custom_worker_group_iam_instance_profile.*.id), count.index)}"
```
Comment (dkelleher) [inline workers.tf]: I assume this is my own ignorance but in order to get this to work for me I had to change this line to
```
iam_instance_profile  = "${element(coalescelist(aws_iam_instance_profile.workers.*.id, data.aws_iam_instance_profile.custom_worker_group_iam_instance_profile.*.name), count.index)}"
```
This appears to be in line with description of the parameter in hashicorp's docs.

Comment (erks) [reply workers.tf]: you're actually right. i had `name` in the launch template version below and only tested that version. fixing.

Comment (erks) [reply workers.tf]: thanks. fixed.

**Comment by erks** [comment]
just rebased from master again.

**Comment by DTTerastar** [comment]
> Downloaded module for testing purposes, was able to successfully spin up EKS clusters with custom IAM permissions using this PR.

Me too!

**Comment by max-rocket-internet** [comment]
I just rebased this so I can test it now...

**Comment by max-rocket-internet** [comment]
OK, since we have a couple of people who have tested it and I also did, I think we merge 🙂

I do think that we are really approaching the limits of Terraform complexity now. It's quite hard to follow some of this code. I just hope Terraform 0.12 can help claw back some readability and simplicity soon 😅
